grace/utils/augment_graph.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Dict, Union, Tuple
+
+import torch
+
+import numpy as np
+
+from grace.base import GraphAttrs, Annotation
+
+
+class RandomEdgeAdditionAndRemoval:
+    """Rotate the image and graph in tandem; i.e., the graph x-y coordinates
+    will be transformed to reflect the image rotation.
+
+    Accepts an image stack of size (C,W,H) or (B,C,W,H) and a dictionary
+    which includes the graph object.
+
+    Parameters
+    ----------
+    n_add : int
+        Number of new edges to add
+    n_remove : Tuple[int]
+        Number of existing edges to remove
+    rng : numpy.random.Generator
+        Random number generator
+    """
+
+    def __init__(
+        self,
+        n_add: int = 10,
+        n_remove: int = 10,
+        rng: np.random.Generator = np.random.default_rng(),
+    ):
+        self.rng = rng
+        self.n_add = n_add
+        self.n_remove = n_remove
+
+    def __call__(
+        self, x: torch.Tensor, graph: Dict[Union[str, GraphAttrs], Any]
+    ) -> Tuple[torch.Tensor, dict]:
+        edges_list = list(graph["graph"].edges)
+        max_node = graph["graph"].number_of_nodes()
+        edges_to_add = [
+            tuple(self.rng.integers(low=0, high=max_node, size=(2,)))
+            + ({GraphAttrs.EDGE_GROUND_TRUTH: Annotation.UNKNOWN},)
```

Comment:
One option here we could have is to randomly assign a label to the edge, something like:

```python
edge_label = rng.choice(Annotation)
```

Reply:
Although maybe it's good to leave it as is - I'm not sure what's best TBH

Reply:
ideally, could the edge annotation be a sort of average of the ones close to where the new edge is?

Reply:
Again, I might be wrong here, but I feel like having it be random, adds to the augmentation aspect

Reply:
@mooniean @quantumjot I've implemented something along these lines with the `find_average_annotation ` function, let me know if this looks alright.

Reply:
I think this should be optional. The default should be either random annotation or `UNKNOWN`